### PR TITLE
PWN-8271 - show claim bundle progress details on tap on claim cell on home screen

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8271

## Description of Work

PWN-8271 - show claim bundle progress details on tap on claim cell on home screen

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/bridge/api/response/BridgeBundleResponse.kt [hunk 1]
```diff
@@ -32,4 +33,6 @@ class BridgeBundleResponse(
     val status: ClaimStatus? = null,
     @SerializedName("claim_key")
     val claimKey: String? = null,
+    @SerializedName("created")
+    val dateCreated: MillisSinceEpoch
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/api/response/BridgeBundleResponse.kt:37]: dateCreated is not millis

File: app/src/main/java/org/p2p/wallet/bridge/claim/repository/EthereumBridgeInMemoryRepository.kt [hunk 2]
```diff
@@ -32,6 +34,16 @@ class EthereumBridgeInMemoryRepository : EthereumBridgeLocalRepository {
...
     }
 
+    override fun getBundleByToken(token: Token.Eth): BridgeBundle? {
+        return if (token.isEth) {
+            bridgeBundlesMap.values.filter { it.status.isProcessing() }
+                .lastOrNull() { it.resultAmount.token == null }
+        } else {
+            bridgeBundlesMap.values.filter { it.status.isProcessing() }
+                .lastOrNull() { token.publicKey == it.resultAmount.token?.hex }
+        }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/claim/repository/EthereumBridgeInMemoryRepository.kt:44]:
```suggestion
        return bridgeBundlesMap.values.filter { it.status.isProcessing() }
            .lastOrNull() { 
                    if (token.isEth) {
                        it.resultAmount.token == null
                   } else { 
                       token.publicKey == it.resultAmount.token?.hex
                   }
           }
```

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt [hunk 3]
```diff
@@ -217,17 +221,54 @@ class HomePresenter(
...
+                    }
+                } ?: tryGetProgressByToken(token)
             }
         }
     }
 
+    private suspend fun tryGetProgressByToken(token: Token.Eth) {
+        ethereumInteractor.getBundleByToken(token)?.let { bridgeBundle ->
+            getProgressDetails(token, bridgeBundle)
+        } ?: Timber.i("No bundle found associated to token: $token to show claim details!")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:243]: should we log as .e?

**Review by gslevinkov — APPROVED** [review]

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/bridge/api/response/BridgeBundleResponse.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/bridge/api/response/BridgeBundleResponse.kt:37]: and what do you suggest?

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt (hunk 3 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:243]: maybe
i do not know how much this is important
but on lets do it with `.e`

File: app/src/main/java/org/p2p/wallet/bridge/api/response/BridgeBundleResponse.kt (hunk 1 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/bridge/api/response/BridgeBundleResponse.kt:37]: making just Long and maybe add sec to name

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt (hunk 3 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:243]: TImber.e(IllegalStateException("No bundle found associated to token: $token to show claim details!")) will be ok for now

File: app/src/main/java/org/p2p/wallet/bridge/api/response/BridgeBundleResponse.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/bridge/api/response/BridgeBundleResponse.kt]: fixed

**Review by Davran331122 — APPROVED** [review]

**Pull request merged**